src/contexts/data/categoryContext.js

```diff
@@ -0,0 +1,44 @@
+import axios from 'axios';
+import {
+  createContext,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
+
+const CategoryContext=createContext();
+
+const CategoryProvider=({children})=>{
```

Comment:
if you want you can but you can also avoid using the categories API by using the products you fetch and very specific you can call the category on that page itself